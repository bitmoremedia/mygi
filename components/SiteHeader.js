import React from 'react';

import { config } from 'config';
import PageLink from './common/PageLink';

const SiteHeader = ({ activePath }) => {

  const gylcemicIndexPathActive = (activePath.indexOf("/categories/") > -1 || activePath === "/");
  const aboutPathActive = (activePath.indexOf("/about") > -1);

  return (
    <div className='site-header'>
      <PageLink to={'/'}>
        <h1>
          MyGi
        </h1>
      </PageLink>
      <PageLink to={'/'}>
        <div>
          Gylcemic Index {gylcemicIndexPathActive && '*'}
        </div>
      </PageLink>
      <PageLink to={'/about'}>
        <div>
          About {aboutPathActive && '*'}
        </div>
      </PageLink>
    </div>
  );
};

export default SiteHeader;
