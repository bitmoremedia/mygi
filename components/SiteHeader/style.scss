@import '../../css/variables';

.site-header {
  background-color: $main;
  height: 60px;
}
.site-header__home-link {
  font-size: 25px;
  height: 60px;
  line-height: 60px;
  text-align: center;
  @include respond-to(tablet) {
    font-size: 30px;
  }
}
.site-header__home-link,
.site-header__home-link:hover,
.site-header__home-link:focus {
  color: $textOnMain;
}
.site-header__logo-image {
  height: 46px;
  border-radius: 9px;
  margin-top: 7px;
}

.site-header__page-links {
  float: right;
}
.site-header__page-link-container {
  float: left;
}
.site-header__page-link {
  height: 60px;
  line-height: 60px;
  text-align: center;
  margin-left: 10px;
  margin-right: 10px;
  @include respond-to(mobile-large) {
    margin-left: 20px;
    margin-right: 20px;
  }
  @include respond-to(tablet) {
    margin-left: 40px;
    margin-right: 40px;
  }
}
.site-header__page-link,
.site-header__page-link:hover,
.site-header__page-link:focus {
  color: $textOnMain;
}
.site-header__page-link--active,
.site-header__page-link--active:hover,
.site-header__page-link--active:focus {
  border-bottom: solid 1px $textOnMain;
  display: inline;
  padding-bottom: 3px;
}
