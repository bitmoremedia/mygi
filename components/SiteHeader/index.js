import React, { PropTypes } from 'react'
import classNames from 'classnames'

import './style.scss'
import PageLink from '../common/PageLink'
import SiteContent from '../common/SiteContent'
import logoImage from '../../img/mygi-app-icon.png'

const SiteHeader = ({ activePath }) => {
  const giPathActive = (activePath.indexOf('/glycemic-index/') > -1)
  const aboutPathActive = (activePath.indexOf('/about') > -1)

  const giLinkClass = classNames('site-header__page-link-container', { 'site-header__page-link-container--active': giPathActive })
  const aboutLinkClass = classNames('site-header__page-link-container', { 'site-header__page-link-container--active': aboutPathActive })

  return (
    <div className="site-header">
      <SiteContent className="site-content__site-header">
        <PageLink className="site-header__home-link" to={'/'}>
          <img className="site-header__logo-image" src={logoImage} alt="MyGi Logo" />
        </PageLink>
        <div className="site-header__page-links">
          <PageLink className={giLinkClass} to={'/glycemic-index'}>
            <div className="site-header__page-link">
              Glycemic Index
            </div>
          </PageLink>
          <PageLink className={aboutLinkClass} to={'/about'}>
            <div className="site-header__page-link">
              About
            </div>
          </PageLink>
        </div>
      </SiteContent>
    </div>
  )
}

SiteHeader.propTypes = {
  activePath: PropTypes.string,
}

SiteHeader.defaultProps = {
  activePath: '',
}

export default SiteHeader
