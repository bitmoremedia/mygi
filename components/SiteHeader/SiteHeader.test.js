import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import SiteHeader from './index'

test('component renders without error', () => {
  const wrapper = shallow(
    <SiteHeader />,
  )
  expect(wrapper.exists()).toBe(true)
})

test('component renders as expected', () => {
  const component = renderer.create(
    <SiteHeader />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('about page active', () => {
  const component = renderer.create(
    <SiteHeader activePath="/about" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('glycemic index page active', () => {
  const component = renderer.create(
    <SiteHeader activePath="/glycemic-index/" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('glycemic index sub page active', () => {
  const component = renderer.create(
    <SiteHeader activePath="/glycemic-index/beans/" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
