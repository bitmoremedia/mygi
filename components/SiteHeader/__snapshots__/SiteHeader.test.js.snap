// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about page active 1`] = `
<div
  className="site-header"
>
  <div
    className="site-content site-content__site-header"
  >
    <div
      className="site-content__inner"
    >
      <a
        className="site-header__home-link"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="MyGi Logo"
          className="site-header__logo-image"
          src="mygi-app-icon.png"
        />
      </a>
      <div
        className="site-header__page-links"
      >
        <a
          className="site-header__page-link-container"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            Glycemic Index
          </div>
        </a>
        <a
          className="site-header__page-link-container site-header__page-link-container--active"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            About
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`component renders as expected 1`] = `
<div
  className="site-header"
>
  <div
    className="site-content site-content__site-header"
  >
    <div
      className="site-content__inner"
    >
      <a
        className="site-header__home-link"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="MyGi Logo"
          className="site-header__logo-image"
          src="mygi-app-icon.png"
        />
      </a>
      <div
        className="site-header__page-links"
      >
        <a
          className="site-header__page-link-container"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            Glycemic Index
          </div>
        </a>
        <a
          className="site-header__page-link-container"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            About
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`glycemic index page active 1`] = `
<div
  className="site-header"
>
  <div
    className="site-content site-content__site-header"
  >
    <div
      className="site-content__inner"
    >
      <a
        className="site-header__home-link"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="MyGi Logo"
          className="site-header__logo-image"
          src="mygi-app-icon.png"
        />
      </a>
      <div
        className="site-header__page-links"
      >
        <a
          className="site-header__page-link-container site-header__page-link-container--active"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            Glycemic Index
          </div>
        </a>
        <a
          className="site-header__page-link-container"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            About
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`glycemic index sub page active 1`] = `
<div
  className="site-header"
>
  <div
    className="site-content site-content__site-header"
  >
    <div
      className="site-content__inner"
    >
      <a
        className="site-header__home-link"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="MyGi Logo"
          className="site-header__logo-image"
          src="mygi-app-icon.png"
        />
      </a>
      <div
        className="site-header__page-links"
      >
        <a
          className="site-header__page-link-container site-header__page-link-container--active"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            Glycemic Index
          </div>
        </a>
        <a
          className="site-header__page-link-container"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="site-header__page-link"
          >
            About
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
